### DISCUSSION TITLE
Go-segment not working well with Goland?

### DISCUSSION THREAD

**Discussion by Hultan** [JanDeDobbeleer/oh-my-posh#1420] [Troubleshoot] [answered]
Hi,

I am using **JetBrains Goland** when developing my Go-applications, and I have updated all my projects to 1.17. However, when I started to use **oh-my-posh** yesterday, I noticed that it reported that I had version 1.14 of Go.

I might very well have version 1.14 installed on my machine since way back, but I am 100% sure that pretty much all of my go.mod files have 1.17. I think what happen here is that **Goland** installs Go at ~/sdk/go1.17.x instead of /usr/local/go/bin, so the version I have at /usr/local/go/bin might very well be 1.14 as **oh-my-posh** reports. If I type **go version** in the terminal, it does report 1.14.

It might be better to read the version from the go.mod file in the current dir instead? 

This might be related to #165, but that issue is closed, so maybe I have misunderstood something from the docs, and **oh-my-posh** do use the go.mod version with the right settings?

/Per

**Comment by JanDeDobbeleer** [comment, marked as answer]
It indeed does go --version and imho that's how it should be. We can however add functionality to parse the mod file too (we do that for angular too).

**Reply by Hultan** [reply]
I understand, but I do think that the version that **go version** and **oh-my-posh** reports, are pretty useless for **Goland** users, that don't really care what version is installed in /usr/local/go/bin. I didn't even realize that I had an old version on my machine until I started to use **oh-my-posh**.

So I would love to have a way to tell **oh-my-posh** to look in the go.mod file instead. Maybe a **prefer-go-mod-version** option?

**Reply by JanDeDobbeleer** [reply]
@Hultan we can facilitate that as I stated already. Stay tuned for an update.

**Reply by JanDeDobbeleer** [reply]
@Hultan see #1428 (in progress)

**Reply by JanDeDobbeleer** [reply]
@Hultan it's live

**Comment by Hultan** [comment]
Thanks! Works great :-)
